Index: python-oq/openquake/bin/cache_gc.py
===================================================================
--- python-oq.orig/openquake/bin/cache_gc.py
+++ python-oq/openquake/bin/cache_gc.py
@@ -44,11 +44,8 @@ LONG_ARGS = ['help', 'job=', 'list']
 S2L = dict(h='help', j='job', l='list')
 
 
-def main(cl_args):
-    """
-    :param cl_args: command line arguments
-    :type cl_cargs: list of strings
-    """
+def main():
+    cl_args = sys.argv[1:]
     try:
         opts, _ = getopt.getopt(cl_args, SHORT_ARGS, LONG_ARGS)
     except getopt.GetoptError, e:
@@ -153,4 +150,4 @@ def show_help():
 
 
 if __name__ == '__main__':
-    main(sys.argv[1:])
+    main()
