--- a/bin/openquake_messages_collector.py
+++ b/bin/openquake_messages_collector.py
@@ -8,21 +8,17 @@
 in logging.cfg
 """
 
-import os
-
 import logging
 import logging.config
 
 import oqpath
 oqpath.set_oq_path()
 
-import openquake
 from openquake import signalling
 
 
 def main():
-    logging.config.fileConfig(os.path.join(openquake.OPENQUAKE_ROOT,
-                                           'logging.cfg'))
+    logging.config.fileConfig("/etc/openquake/logging.cfg")
 
     # any job
     collector = signalling.Collector('*', logging.getLogger('collector'))
