Index: oq-engine/openquake/engine/openquake.cfg
===================================================================
--- a/openquake/engine/openquake.cfg
+++ b/openquake/engine/openquake.cfg
@@ -38,7 +38,7 @@
 
 [dbserver]
 user = openquake
-file = ~/oqdata/db.sqlite3
+file = /var/lib/openquake/db.sqlite3
 # daemon bind address; must be a valid IP address
 # example: 0.0.0.0
 listen = 127.0.0.1
@@ -47,9 +47,10 @@
 # of the master node (on the master node cfg too)
 # example: master.hpc
 host = local
-# port 1908 has a good reputation:
-# https://isc.sans.edu/port.html?port=1908
-port = 1908
+# for packages we use port 1907 to avoid conflicts
+# with local development installations
+# https://isc.sans.edu/port.html?port=1907
+port = 1907
 # receiver host; if missing use hostname
 receiver_host = 
 # port range used by workers to send back results
 # to the master node
 receiver_ports = 1912-1920
