Index: oq-engine/openquake/engine/openquake.cfg
===================================================================
--- oq-engine.orig/openquake/engine/openquake.cfg	2018-03-15 08:47:12.245368066 +0200
+++ oq-engine/openquake/engine/openquake.cfg	2018-03-15 08:47:12.245368066 +0200
@@ -39,15 +39,14 @@
 celery_queue = celery
 
 [dbserver]
-# enable multi_user if you have a multiple user installation
-multi_user = false
-file = ~/oqdata/db.sqlite3
-log = ~/oqdata/dbserver.log
+multi_user = true
+file = /var/lib/openquake/db.sqlite3
+log = /var/lib/openquake/dbserver.log
 host = 127.0.0.1
-# port 1908 has a good reputation:
-# https://isc.sans.edu/port.html?port=1908
-port = 1908
-authkey = changeme
+# for packages we use port 1907 to avoid conflicts
+# with local development installations
+# https://isc.sans.edu/port.html?port=1907
+port = 1907
 
 [zworkers]
 master_host = 127.0.0.1

