Index: oq-engine/openquake/engine/openquake.cfg
===================================================================
--- oq-engine.orig/openquake/engine/openquake.cfg	2016-05-06 14:04:42.073467919 +0000
+++ oq-engine/openquake/engine/openquake.cfg	2016-05-06 14:05:35.566145165 +0000
@@ -41,7 +41,7 @@
 [dbserver]
 # enable multi_user if you have a multiple user installation
 multi_user = false
-file = ~/db.sqlite3
+file = /var/lib/openquake/db.sqlite3
 host = localhost
 port = 1999
 authkey = changeme
