Index: oq-engine/openquake/engine/openquake.cfg
===================================================================
--- oq-engine.orig/openquake/engine/openquake.cfg	2017-10-13 07:49:34.110383342 +0200
+++ oq-engine/openquake/engine/openquake.cfg	2017-10-13 08:24:07.242368866 +0200
@@ -40,9 +40,9 @@
 
 [dbserver]
 # enable multi_user if you have a multiple user installation
-multi_user = false
-file = ~/oqdata/db.sqlite3
-log = ~/oqdata/dbserver.log
+multi_user = true
+file = /var/lib/openquake/db.sqlite3
+log = /var/lib/openquake/dbserver.log
 host = 127.0.0.1
 # port 1908 has a good reputation:
 # https://isc.sans.edu/port.html?port=1908
