Index: oq-engine/openquake.cfg
===================================================================
--- oq-engine.orig/openquake.cfg	2016-05-06 14:06:06.000000000 +0000
+++ oq-engine/openquake.cfg	2016-05-06 16:23:04.006138759 +0000
@@ -40,9 +40,9 @@
 
 [dbserver]
 # enable multi_user if you have a multiple user installation
-multi_user = false
-file = ~/oqdata/db.sqlite3
-log = ~/oqdata/dbserver.log
+multi_user = true
+file = /var/lib/openquake/oqdata/db.sqlite3
+log = /var/lib/openquake/oqdata/dbserver.log
 host = localhost
 port = 1999
 authkey = changeme
