--- a/bin/oq-restart
+++ b/bin/oq-restart
@@ -11,7 +11,7 @@
 
 if (ps ax | grep "openquake" | egrep --invert-match --silent "grep|postgres") ; then
 	echo "it is."
-	echo "Please stop all openquake calculations before running $0"
+	echo "Please stop all openquake jobs before running $0"
 	exit 2
 fi
 
@@ -28,12 +28,12 @@
 	echo "Celeryd log file has been moved to /tmp/celeryd.log.old"
 fi
 
-echo -n "Updating terminated calculations status... "
+echo -n "Updating terminated jobs' status... "
 
 PGPASSWORD=openquake psql -U oq_job_init -h localhost openquake <<EOF
-	UPDATE uiapi.oq_calculation
+	UPDATE uiapi.oq_job
 	SET status = 'failed'
-	WHERE status NOT IN ('succeded', 'failed');
+	WHERE status NOT IN ('succeeded', 'failed');
 EOF
 
 export MPLCONFIGDIR=/tmp
@@ -51,7 +51,6 @@
 /etc/init.d/rabbitmq-server restart
 /etc/init.d/redis-server restart
 
-/usr/bin/oq-queue-delete celery
 echo -n "Starting $concurrency celery workers for OpenQuake... "
 
 celeryd -c $concurrency --purge > /tmp/celeryd.log 2>&1 3>&1 &
