Index: python-oq/openquake/bin/create_oq_schema
===================================================================
--- python-oq.orig/openquake/bin/create_oq_schema	2012-09-01 07:25:23.554757903 +0200
+++ python-oq/openquake/bin/create_oq_schema	2012-09-04 08:40:24.685310876 +0200
@@ -26,7 +26,7 @@
 fi
 
 # Openquake version
-oq_version="0.5.0"
+oq_version="0.8.4"
 
 # Stop at the first error in a batch to give a chance to see the error message
 psql_batch_options='--set ON_ERROR_STOP=1'
Index: python-oq/openquake/__init__.py
===================================================================
--- python-oq.orig/openquake/__init__.py	2012-09-01 07:25:23.554757903 +0200
+++ python-oq/openquake/__init__.py	2012-09-04 08:42:00.362046349 +0200
@@ -48,16 +48,11 @@
 along with OpenQuake.  If not, see <http://www.gnu.org/licenses/>.
 """
 
-import os
-
 # Please note: the release date should always have a value of 0 (zero) in the
 # master branch. It will only be set to a meaningful value in *packaged* and
 # released OpenQuake code.
 __version__ = (
     0,  # major
     8,  # minor
-    1,  # sprint number
+    4,  # sprint number
     0)  # release date (seconds since the "Epoch"), do *not* set in master!
-
-# The path to the OpenQuake root directory
-OPENQUAKE_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
